lib/config/flat-config-schema.js

```diff
@@ -126,32 +126,65 @@ function normalizeRuleOptions(ruleOptions) {
 // Assertions
 //-----------------------------------------------------------------------------
 
+/**
+ * The error type when a rule's options are configured with an invalid type.
+ */
+class InvalidRuleOptionsError extends Error {
+
+    /**
+     * @param {string} ruleId Rule name being configured.
+     * @param {any} value The invalid value.
+     */
+    constructor(ruleId, value) {
+        super(`Key "${ruleId}": Expected severity of "off", 0, "warn", 1, "error", or 2.`);
```

Comment:
🤔 is there a specific situation where the `string, number, or array` messaging is better? I would have thought it friendlier for users to explicitly tell them what severities are allowed? As a new user, I imagine it would be useful to be told more specifically.

Reply:
I was thinking about cases like `"prefer-const": { destructuring: true }` where the problem is invalid type rather than invalid severity, but now when reading the new message again I see that it covers missing severity as well as invalid severity, so it sounds good to me.